lib/tests/unit/connector/connector_test.cc

```diff
@@ -162,113 +207,76 @@ TEST_CASE("Connector::startMonitoring", "[connector]") {
+                    connected = true;
                 });
-        mock_server.set_ping_handler(
-                [&num_pings](websocketpp::connection_hdl hdl, std::string) -> bool {
-                    ++num_pings;
-                    return false;
+        mock_server_ptr->set_ping_handler(
+                [](websocketpp::connection_hdl hdl, std::string) -> bool {
+                    return true;
                 });
-        mock_server.go();
-        auto port = mock_server.port();
+        mock_server_ptr->go();
+        auto port = mock_server_ptr->port();
 
-        // Setting the pong timeout to 500 ms and the Association
-        // timeout to 1 s (Connector::connect will hang waiting for
-        // for the Association completion that won't happen because
-        // we'll be done as soon as num_connections > 1)
+        // Setting pong timeout to 500 ms and Association timeout to 1 s
         Connector c { "wss://localhost:" + std::to_string(port) + "/pcp",
                       "test_client",
                       getCaPath(), getCertPath(), getKeyPath(),
-                      WS_TIMEOUT_MS, 1, ASSOCIATION_REQUEST_TTL_S, 1, 500 };
+                      900, 1, ASSOCIATION_REQUEST_TTL_S, 1 };
 
-        REQUIRE(num_connections == 0);
-        REQUIRE(num_pings == 0);
+        REQUIRE_FALSE(connected);
         REQUIRE_NOTHROW(c.connect(1));
         REQUIRE(c.isConnected());
-        REQUIRE(num_connections == 1);
 
-        // Infinite retries; 1 s between each WebSocket ping
-        Util::thread monitor { [&]() {
-            REQUIRE_NOTHROW(c.monitorConnection(0, 1));
-        } };
+        // Monitor with 1 connection retry and 1 s between each WebSocket ping
+        if (use_blocking_call) {
+            monitor = Util::thread(
+                    [&]() {
+                        REQUIRE_THROWS_AS(c.monitorConnection(1, 1),
+                                          connection_fatal_error);
+                    });
+        } else {
+            REQUIRE_NOTHROW(c.startMonitoring(1, 1));
+        }
+
+        // Destroy the broker
+        mock_server_ptr.reset(nullptr);
 
-        // After 1 pong timeout, the Connector should close and reconnect
-        wait_for([&num_connections](){return num_connections > 1;}, 30);
+        // Wait for the connection to drop plus 3 seconds to let the monitor
+        // task fail (monitoring period is 1 s)
+        wait_for([&c](){return !c.isConnected();});
+        Util::this_thread::sleep_for(Util::chrono::seconds(3));
 
-        REQUIRE(num_connections > 1);
-        REQUIRE(num_pings > 0);
+        if (!use_blocking_call)
+            REQUIRE_THROWS_AS(c.stopMonitoring(), connection_fatal_error);
 
-        REQUIRE_NOTHROW(c.stopMonitoring());
-        // Ensure monitoring halts completely before we attempt to destroy the connection.
-        // Otherwise destruction will attempt to cleanup the object before startMonitorTask
```

Comment:
Don't want to keep this comment? I thought it helped clarify why we need to join.